nomad/state/state_store.go

```diff
@@ -1404,12 +1493,78 @@ func (r *StateRestore) PeriodicLaunchRestore(launch *structs.PeriodicLaunch) err
 
 // JobSummaryRestore is used to restore a job summary
 func (r *StateRestore) JobSummaryRestore(jobSummary *structs.JobSummary) error {
-	if err := r.txn.Insert("job_summary", jobSummary); err != nil {
+	if err := r.txn.Insert("job_summary", *jobSummary); err != nil {
 		return fmt.Errorf("job summary insert failed: %v", err)
 	}
 	return nil
 }
 
+// CreateJobSummaries computes the job summaries for all the jobs
+func (r *StateRestore) CreateJobSummaries() error {
+	// Get all the jobs
+	var jobs []*structs.Job
+	iter, err := r.txn.Get("jobs", "id")
+	if err != nil {
+		return fmt.Errorf("couldn't retrieve jobs: %v", err)
+	}
+	for {
+		raw := iter.Next()
+		if raw == nil {
+			break
+		}
+		jobs = append(jobs, raw.(*structs.Job))
+	}
+
+	for _, job := range jobs {
+
+		// Get all the allocations for the job
+		iter, err = r.txn.Get("allocs", "job", job.ID)
+		if err != nil {
+			return fmt.Errorf("couldn't retrieve allocations for job %v: %v", job.ID, err)
+		}
+		var allocs []*structs.Allocation
+		for {
+			raw := iter.Next()
+			if raw == nil {
+				break
+			}
+			allocs = append(allocs, raw.(*structs.Allocation))
+		}
+
```

Comment:
We should only create the summary if it doesn't already exist for the job, otherwise we will unnecessarily over-write the existing values